beacon/validator/block_builder.go

```diff
@@ -175,33 +175,23 @@ func (s *Service) getEmptyBeaconBlockForSlot(
 	)
 }
 
-func (s *Service) buildForkData(
-	st *statedb.StateDB,
-	slot math.Slot,
-) (*ctypes.ForkData, error) {
-	var (
-		epoch = s.chainSpec.SlotToEpoch(slot)
-	)
-
+func (s *Service) buildForkData(st *statedb.StateDB, slot math.Slot) (*ctypes.ForkData, error) {
 	genesisValidatorsRoot, err := st.GetGenesisValidatorsRoot()
 	if err != nil {
 		return nil, err
 	}
 
 	return ctypes.NewForkData(
-		bytes.FromUint32(s.chainSpec.ActiveForkVersionForEpoch(epoch)), genesisValidatorsRoot,
+		bytes.FromUint32(s.chainSpec.ActiveForkVersionForSlot(slot)), genesisValidatorsRoot,
```

Comment:
@calbera : why this change?

Reply:
there was a conversion from slot to epoch happening above. its not necessary since we already have this function that just uses slot on chainSpec